internal/api/attestationconfigapi/reporter.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+/*
+The reporter contains the logic to determine a latest version for Azure SEVSNP based on cached version values observed on CVM instances.
+*/
+package attestationconfigapi
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/edgelesssys/constellation/v2/internal/api/client"
+	apiclient "github.com/edgelesssys/constellation/v2/internal/api/client"
+	"github.com/edgelesssys/constellation/v2/internal/attestation/variant"
+)
+
+// cachedVersionsSubDir is the subdirectory in the bucket where the cached versions are stored.
+const cachedVersionsSubDir = "cached-versions"
+
+// versionWindowSize defines the number of versions to be considered for the latest version. Each week 5 versions are uploaded for each node of the verify cluster.
+const versionWindowSize = 15
+
+var reportVersionDir = path.Join(attestationURLPath, variant.AzureSEVSNP{}.String(), cachedVersionsSubDir)
+
+// UpdateLatestAzureSEVSNPVersion reports the given version, checks the reported version values
+// and updates the latest version of the Azure SEVSNP in the API if there is an update .
+// force can be used to force an update of the latest API version regardless of the cache reporting.
+func (c Client) UpdateLatestAzureSEVSNPVersion(ctx context.Context, inputVersion,
```

Comment:
```suggestion
func (c Client) UploadAzureSEVSNPVersionLatest(ctx context.Context, inputVersion,
```

What we should have done from the beginning: have a AzureSEVSNPVersionLatest object. But since we didn't I think we should at least stick to our established naming.